libwallet/assets/dcr/ticket.go

```diff
@@ -123,13 +124,16 @@ func (asset *Asset) PurchaseTickets(account, numTickets int32, vspHost, passphra
 	}
 	defer asset.LockWallet()
 
+	log.Info("Setting the ticket(s) purchasing account info")
+	vspClient.SetAccountInfo(account, account)
```

Comment:
Upstream dcrwallet doesn't modify the fee and change accounts after the vsp is created with a policy that defines those values. All subsequent calls to wallet.PurchaseTickets continue to use the same accounts initially set in the policy, even if the `PurchaseTicketsRequest` uses a different source account.

Modifying that policy everytime this `PurchaseTickets` method is called is potentially problematic. There's nothing to ensure that this method isn't called twice in quick succession, with the callers passing different accounts in each call, and ending up getting unexpected outcomes.

I understand the desire to use the same purchasing account as fee and change account but perhaps go with dcrwallet's style for now and never change those account values once they've been set at the time of creating the vspClient.

Or come up with a safer way to make this value dynamic.

I remember I'd discussed this issue with the upstream repo maintainer before, even opened a PR to resolve it then.
https://github.com/decred/dcrwallet/pull/2129

Reply:
Our implementation here reuses the VSP clients therefore we cannot set the fee and change accounts when creating a client.
`There's nothing to ensure that this method isn't called twice in quick succession` theoretically this is possible but how `vspClient.SetAccountInfo(` is used in the code, that case scenario is impossible to replicate. I feel adding those extra safeguards will only make the method more complicated and error prone.

[As mentioned earlier](https://github.com/crypto-power/cryptopower/pull/48#discussion_r1307687402):
>..this account information is set before the ticket buying loop starts, so we don't have any scenario of multiple synchronous writes to those variables would happen...

That account information is set only once together with other ticket information and it doesn't happen more than one per given ticket purchase trigger. Only the following methods set the account information:
- `func (asset *Asset) PurchaseTickets(account, numTickets int32, vspHost, passphrase string, vspPubKey []byte) ([]*chainhash.Hash, error) {`
- `func (asset *Asset) runTicketBuyer(ctx context.Context, passphrase string, cfg *TicketBuyerConfig) error {`

None of them can execute in multiple quick successions and if it does, this is a bug where that is triggered.